crates/chain-config/src/config/contract.rs

```diff
@@ -28,6 +28,9 @@ use serde_with::{
 #[serde_as]
 #[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
 pub struct ContractConfig {
+    #[serde_as(as = "HexType")]
+    #[serde(default)]
+    pub contract_id: ContractId,
```

Comment:
Should we make `contract_id` optional and automatically calculate it as we did before if it's set to None?

Reply:
The primary purpose of the `ContractConfig` is the re-genesis of the blockchain, so I would prefer to keep it as a required field because each contract has a contract id used during initial creation.